CRM/Core/Page/AJAX.php

```diff
@@ -108,21 +108,19 @@ public static function setIsQuickConfig() {
     }
     $priceSetId = CRM_Price_BAO_PriceSet::getFor($context, $id, NULL);
     if ($priceSetId) {
-      $result = CRM_Price_BAO_PriceSet::setIsQuickConfig($priceSetId, 0);
+      $sql = "UPDATE
+       civicrm_price_set cps
+       INNER JOIN civicrm_price_set_entity cpse ON cps.id = cpse.price_set_id
+       INNER JOIN {$context} ce ON cpse.entity_id = ce.id AND ce.id = %1
+       SET cps.is_quick_config = 0, cps.financial_type_id = IF(cps.financial_type_id IS NULL, ce.financial_type_id, cps.financial_type_id)
+      ";
+      CRM_Core_DAO::executeQuery($sql, array(1 => array($id, 'Integer')));
+
       if ($context == 'civicrm_event') {
-        $sql = "UPDATE
```

Comment:
I'm nervous about removing this query - I think it would do something useful if an early bird discount is configured....

Reply:
And I was wondering if the discount table might be configured with a different price-set then event because as per UI it's not possible. And if that's true then it's safe as the because the update query does the same job as earlier. What you think?

Reply:
@monishdeb I just tested now what happens before merging this change if you configure an early discount price set and then convert the price set.

The result is that 2 price sets now exist - one for the discount and that one has a correct financial type but the start & end dates are not carried over to the price_field table and since there are now 2 price sets only one can be assigned.

I guess it fails to work badly enough that I think I think it is OK to remove the attempt to convert the early bird discount rate. I think the CORRECT fix (but out of scope) would be to insert those rows into the main price set with the dates. However, there is some complexity then as the main price set fields would not be date dependent and would also show. Arguably that is a good thing (when I have done events with early bird pricing I usually make both know so people can see they are getting a discount). 

Upshot is this does fix the problem, and it alters another problem that no-one has noticed. I'm going to merge it.